src/sinks/azure_common/config.rs

```diff
@@ -121,34 +121,44 @@ pub fn build_client(
     connection_string: Option<String>,
     storage_account: Option<String>,
     container_name: String,
+    blob_endpoint: Option<String>,
 ) -> crate::Result<Arc<ContainerClient>> {
     let client;
     match (connection_string, storage_account) {
         (Some(connection_string_p), None) => {
             let connection_string = ConnectionString::new(&connection_string_p)?;
-            let mut builder = ClientBuilder::new(
-                connection_string
-                    .account_name
-                    .ok_or("Account name missing in connection string")?,
-                connection_string.storage_credentials()?,
-            )
-            .retry(RetryOptions::none());
-            if let Some(uri) = connection_string.blob_endpoint {
-                builder = builder.cloud_location(CloudLocation::Custom {
+
+            let builder = match connection_string.blob_endpoint {
+                Some(uri) => ClientBuilder::with_location(CloudLocation::Custom {
                     uri: uri.to_string(),
                     credentials: connection_string.storage_credentials()?,
-                });
+                }),
+                None => ClientBuilder::new(
+                    connection_string
+                        .account_name
```

Comment:
When a `blob_endpoint` is specified in the connection string, the `account_name` does not need to be specified in the connection string.